Terminal.Gui/Views/Menu.cs

```diff
@@ -76,6 +78,20 @@ public MenuItem (ustring title, string help, Action action)
 		/// </summary>
 		/// <value>Method to invoke.</value>
 		public Action Action { get; set; }
+
+		/// <summary>
+		/// Gets or sets the action to be invoked if the menu can be triggered
+		/// </summary>
+		/// <value>Function to determine if action is ready to be executed.</value>
+		public Func<bool> CanExecute { get; set; }
+		
+		/// <summary>
+		/// Shortcut to check if the menu item is enabled
+		/// </summary>
+		public bool IsEnabled() {
```

Comment:
Since we need some changes in this patch, let me suggest that you get the indentation to follow the existing code here, the brace should go in a new line.